---
output: github_document
---

```{r setup, include=FALSE}
library(readr)
library(dplyr)
library(knitr)
opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
# theme_set(theme_bw())
# setwd(here::here())
```

## Kintelligence / k2v SNP count comparison

```{r}
sampledata <- read_tsv(here::here("testdata/HG002.SnpResult.txt"), col_types="ciiicciii")
hg19 <- read_tsv(here::here("build/hg19_mapping.tsv"), col_types="cic")
hg19 <- hg19 %>% transmute(chr=Chromosome %>% gsub("23", "X", .) %>% gsub("24", "Y", .), pos=Position, rsid=RsId)
hg19x <- hg19 %>% semi_join(sampledata, by=c("rsid"="Locus_ID"))
alleletable <- read_csv(here::here("build/kintelligence.alleles.csv"), col_types = "cicccd")
notk2v <- hg19x %>% 
  anti_join(alleletable, by=c("chr", "pos", "rsid")) %>% 
  mutate(chr=factor(chr, levels=c(as.character(1:22), "X", "Y")))
```

k2v joins the `<sample>.SnpResult.txt` to an allele table to get genotypes which are used to convert to VCF. 

Building the k2v allele table starts with the [`hg19_mapping.tsv`](hg19_mapping.tsv) file provided by Verogen, which contains RSIDs, chromosome, and position information for `r scales::comma(nrow(hg19))` SNPs, including `r nrow(hg19 %>% filter(chr=="X"))` and `r nrow(hg19 %>% filter(chr=="Y"))` on the X and Y chromosomes, respectively. `r scales::comma(nrow(hg19x))` of these SNPs are genotyped and included in the `<sample>.SnpResult.txt` output, including `r nrow(hg19x %>% filter(chr=="X"))` and `r nrow(hg19x %>% filter(chr=="Y"))` on the X and Y chromosomes, respectively.

The allele table is created by extracting biallelic SNPs from GnomAD genomes for 1-22 and X using the GRCh37 GnomAD site VCF. Y chromosome sites from GnomAD are not available on GRCh37, so the GRCh38 Y chromosome site VCF from GnomAD is lifted over to GRCh37 prior to extracting Y chromosome kintelligence sites. Reference and alternate alleles are extracted from the GnomAD site VCF(s) and used for converting 0/0, 0/1, 1/1 genotypes in the `<sample>.SnpResult.txt`.

Not all of the sites in `hg19_mapping.tsv` and provided in `<sample>.SnpResult.txt` are in GnomAD by both position and RSID. The table below compares by chromosome sites generated by Kintelligence with what's present in GnomAD (and ultimately output by k2v). `r nrow(notk2v)` SNPs are not captured by k2v.

```{r}
inner_join(hg19x %>% count(chr, name="kintel"),
           hg19x %>% inner_join(alleletable, by=c("chr", "pos", "rsid")) %>% count(chr, name="k2v"),
           by="chr") %>% 
  mutate(dif=kintel-k2v) %>% 
  mutate(chr=factor(chr, levels=c(as.character(1:22), "X", "Y"))) %>% 
  arrange(chr) %>% 
  rename(Chromosome=chr, `Kintelligence count`=kintel, `k2v count`=k2v, Difference=dif) %>% 
  knitr::kable()
```

```{r, include=FALSE}
knitr::knit_exit()
```

The table below shows the `r nrow(notk2v)` SNPs not captured by k2v, because they were not biallelic SNPs in GnomAD by position and RSID.

```{r}
notk2v %>% 
  arrange(chr) %>% 
  knitr::kable()
```

